src/main/java/toolpackage/Parser.java

```diff
@@ -0,0 +1,75 @@
+package toolpackage;
+
+import java.nio.charset.Charset;
+
+import dukepackage.DukeException;
+
+import taskpackage.Deadlines;
+import taskpackage.Events;
+import taskpackage.ToDos;
+
+/**
+ * Represents the parser tool for the
+ * Duke bot. A Parser object is used
+ * by the Duke bot to understand
+ * and execute user commands.
+ */
+public class Parser {
+
+    /**
+     * Parses and executes command given by user.
+     *
+     * @param userInput String given by user.
+     * @param tasks Task list of bot to be modified.
+     * @param ui UI of bot.
+     * @return String Response by Duke bot.
+     * @throws DukeException if there are any issues in the command execution.
+     */
+    public static String parse(String userInput, TaskList tasks, Ui ui) throws DukeException {
+        String[] parsedCommand = userInput.split(" ", 2);
+        String command = parsedCommand[0];
+
+        String task = "";
+        if (parsedCommand.length > 1) {
+            task = parsedCommand[1];
+        }
+
+        if (command.equals("bye")) {
```

Comment:
Agree with the other review that perhaps switch statements would improve readability